src/crypto/index.js

```diff
@@ -68,16 +89,33 @@ const MIN_FORCE_SESSION_INTERVAL_MS = 60 * 60 * 1000;
  *    storage for the crypto layer.
  *
  * @param {RoomList} roomList An initialised RoomList object
+ *
+ * @param {Array} verificationMethods Array of verification methods to use.
+ *    Each element can either be a string from MatrixClient.verificationMethods
+ *    or a class that implements a verification method.
  */
 export default function Crypto(baseApis, sessionStore, userId, deviceId,
-                clientStore, cryptoStore, roomList) {
+    clientStore, cryptoStore, roomList, verificationMethods) {
     this._baseApis = baseApis;
     this._sessionStore = sessionStore;
     this._userId = userId;
     this._deviceId = deviceId;
     this._clientStore = clientStore;
     this._cryptoStore = cryptoStore;
     this._roomList = roomList;
+    this._verificationMethods = {};
+    if (verificationMethods) {
+        for (const method of verificationMethods) {
+            if (typeof method === "string") {
```

Comment:
I added both because passing a string would be the way to use the built-in methods, but applications could define their own verification methods and use them by passing in a class.

Reply:
ah, I see